homeassistant/components/modbus/sensor.py

```diff
@@ -4,98 +4,36 @@
 import logging
 import struct
 
-import voluptuous as vol
-
-from homeassistant.components.sensor import (
-    DEVICE_CLASSES_SCHEMA,
-    PLATFORM_SCHEMA,
-    SensorEntity,
-)
+from homeassistant.components.sensor import SensorEntity
 from homeassistant.const import (
-    CONF_ADDRESS,
     CONF_COUNT,
-    CONF_DEVICE_CLASS,
     CONF_NAME,
     CONF_OFFSET,
-    CONF_SCAN_INTERVAL,
     CONF_SENSORS,
-    CONF_SLAVE,
     CONF_STRUCTURE,
     CONF_UNIT_OF_MEASUREMENT,
 )
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers import config_validation as cv
 from homeassistant.helpers.restore_state import RestoreEntity
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from . import number
 from .base_platform import BasePlatform
 from .const import (
-    CALL_TYPE_REGISTER_HOLDING,
-    CALL_TYPE_REGISTER_INPUT,
     CONF_DATA_TYPE,
-    CONF_HUB,
-    CONF_INPUT_TYPE,
     CONF_PRECISION,
-    CONF_REGISTER,
-    CONF_REGISTER_TYPE,
-    CONF_REGISTERS,
-    CONF_REVERSE_ORDER,
     CONF_SCALE,
     CONF_SWAP,
     CONF_SWAP_BYTE,
-    CONF_SWAP_NONE,
     CONF_SWAP_WORD,
     CONF_SWAP_WORD_BYTE,
-    DATA_TYPE_CUSTOM,
-    DATA_TYPE_FLOAT,
-    DATA_TYPE_INT,
     DATA_TYPE_STRING,
-    DATA_TYPE_UINT,
-    DEFAULT_HUB,
-    DEFAULT_SCAN_INTERVAL,
-    DEFAULT_STRUCT_FORMAT,
     MODBUS_DOMAIN,
 )
 
 PARALLEL_UPDATES = 1
 _LOGGER = logging.getLogger(__name__)
 
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We always leave the schema in, but mark it as non usable:

something like:
PLATFORM_SCHEMA = vol.Schema(
    vol.All(
        # Deprecated in Home Assistant 2021.7
        cv.deprecated(DOMAIN),
        {
            DOMAIN: vol.Schema(

And remove the inner parts.

Reply:
👍🏼